src/cmd/services/m3dbnode/config/fs.go

```diff
@@ -78,11 +88,152 @@ type FilesystemConfiguration struct {
 
 	// ForceIndexSummariesMmapMemory forces the mmap that stores the index lookup bytes
 	// to be an anonymous region in memory as opposed to a file-based mmap.
-	ForceIndexSummariesMmapMemory bool `yaml:"force_index_summaries_mmap_memory"`
+	ForceIndexSummariesMmapMemory *bool `yaml:"force_index_summaries_mmap_memory"`
 
 	// ForceBloomFilterMmapMemory forces the mmap that stores the index lookup bytes
 	// to be an anonymous region in memory as opposed to a file-based mmap.
-	ForceBloomFilterMmapMemory bool `yaml:"force_bloom_filter_mmap_memory"`
+	ForceBloomFilterMmapMemory *bool `yaml:"force_bloom_filter_mmap_memory"`
+}
+
+// Validate validates the Filesystem configuration. We use this method to validate
+// fields where the validator package falls short.
+func (f FilesystemConfiguration) Validate() error {
+	if f.WriteBufferSize != nil && *f.WriteBufferSize < 1 {
+		return fmt.Errorf(
+			"fs writeBufferSize is set to: %d, but must be at least 1",
+			*f.WriteBufferSize)
+	}
+
+	if f.DataReadBufferSize != nil && *f.DataReadBufferSize < 1 {
+		return fmt.Errorf(
+			"fs dataReadBufferSize is set to: %d, but must be at least 1",
+			*f.DataReadBufferSize)
+	}
+
+	if f.InfoReadBufferSize != nil && *f.InfoReadBufferSize < 1 {
+		return fmt.Errorf(
+			"fs infoReadBufferSize is set to: %d, but must be at least 1",
+			*f.InfoReadBufferSize)
+	}
+
+	if f.SeekReadBufferSize != nil && *f.SeekReadBufferSize < 1 {
+		return fmt.Errorf(
+			"fs seekReadBufferSize is set to: %d, but must be at least 1",
+			*f.SeekReadBufferSize)
+	}
+
+	if f.ThroughputLimitMbps != nil && !(*f.ThroughputLimitMbps > 0) {
+		return fmt.Errorf(
+			"fs throughputLimitMbps is set to: %f, but must be larger than 0",
+			*f.ThroughputLimitMbps)
+	}
+
+	if f.ThroughputCheckEvery != nil && !(*f.ThroughputCheckEvery > 0) {
+		return fmt.Errorf(
+			"fs throughputCheckEvery is set to: %d, but must be larger than 0",
+			*f.ThroughputCheckEvery)
+	}
+
+	return nil
+}
+
+// FilePathPrefixOrDefault returns the configured file path prefix if configured, or a
+// default value otherwise.
+func (f FilesystemConfiguration) FilePathPrefixOrDefault() string {
+	if f.FilePathPrefix != nil {
+		return *f.FilePathPrefix
+	}
+
+	return defaultFilePathPrefix
+}
+
+// WriteBufferSizeOrDefault returns the configured write buffer size if configured, or a
+// default value otherwise.
+func (f FilesystemConfiguration) WriteBufferSizeOrDefault() int {
+	if f.WriteBufferSize != nil && *f.WriteBufferSize > 1 {
+		return *f.WriteBufferSize
+	}
+
+	return defaultWriteBufferSize
+}
+
+// DataReadBufferSizeOrDefault returns the configured data read buffer size if configured, or a
+// default value otherwise.
+func (f FilesystemConfiguration) DataReadBufferSizeOrDefault() int {
+	if f.DataReadBufferSize != nil && *f.DataReadBufferSize > 1 {
```

Comment:
Can get rid of these bounds checks now I think?

Reply:
yeah good call